File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    let arr = my_string.split('');
    for(let i of queries){
        let s = i[0]
        let e = i[1]
        while(s < e){
            let temp = arr[s];
            arr[s++] = arr[e];
            arr[e--] = temp;
        }
    }
    return arr.join(''); 
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = my_string;
        for(int i = 0; i < queries.length; i++){
            int s = queries[i][0];
            int e = queries[i][1];

            char[] arr = answer.toCharArray(); 
            while (s < e) {
                char temp = arr[s];
                arr[s++] = arr[e];
                arr[e--] = temp;
            }
            answer = new String(arr); 
        }
        return answer;
    }
}